packages/styled-components/src/utils/isPlainObject.js

```diff
@@ -1,8 +1,8 @@
 // @flow
-import { typeOf } from 'react-is';
 
 export default (x: any): boolean =>
   x !== null &&
   typeof x === 'object' &&
   (x.toString ? x.toString() : Object.prototype.toString.call(x)) === '[object Object]' &&
-  !typeOf(x);
+  /* check for reasonable markers that the object isn't an element for react & preact/compat */
+  !(x.$$typeof && ('_owner' in x) || x.__v);
```

Comment:
This avoids relying on internal/undocumented properties:

```suggestion
  !('props' in x && (x.$$typeof || x.constructor === undefined));
```